corehq/messaging/scheduling/util.py

```diff
@@ -8,3 +8,16 @@ def utcnow():
     Defined here to do patching in tests.
     """
     return datetime.utcnow()
+
+
+def domain_has_reminders(domain):
+    from corehq.apps.data_interfaces.models import AutomaticUpdateRule
+    from corehq.apps.reminders.models import CaseReminderHandler
+    from corehq.messaging.scheduling.models import ScheduledBroadcast, ImmediateBroadcast
+
+    return (
+        AutomaticUpdateRule.domain_has_conditional_alerts(domain) or
+        ScheduledBroadcast.domain_has_broadcasts(domain) or
+        ImmediateBroadcast.domain_has_broadcasts(domain) or
+        CaseReminderHandler.domain_has_reminders(domain)
```

Comment:
the most often that this will be called is during the domain calculations which i believe only get run once a day, and since these tables are all expected to always stay pretty small i think it should be ok